src/ButtonComponent/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { ElementPropTypes } from '@volusion/element-proptypes';
+import { StyleSheet, css } from 'aphrodite';
+import { getStyles } from './getStyles';
+import { ButtonProps, ButtonConfig } from './types';
+import { getConfigClasses } from './getConfigClasses';
+
+const Button = ({
+    buttonStyle,
+    href,
+    text,
+    newWindow,
+    globalSettings
+}: ButtonProps) => {
+    const { volComponentButton } = globalSettings.globalComponents;
+    const { primaryButtonStyles, secondaryButtonStyles } = volComponentButton;
+    const classes: any = StyleSheet.create(getStyles(volComponentButton));
+    const primaryBaseClasses = getConfigClasses(classes, primaryButtonStyles);
+    const secondaryBaseClasses = getConfigClasses(
+        classes,
+        secondaryButtonStyles
+    );
+
+    const getButtonClasses = (role: string): string => {
+        if (role === 'secondary') {
+            return `${secondaryBaseClasses} ${css(
+                classes.secondaryButton,
+                classes.secondaryButtonHover
+            )}`;
+        } else {
+            return `${primaryBaseClasses} ${css(
+                classes.primaryButton,
+                classes.primaryButtonHover
+            )}`;
+        }
+    };
```

Comment:
This chunk of code right here has the most potential for simplification. Instead of generating all of the classes for both styles before we render anything, we can create the stylesheet based only on the relevant global settings, and ignore the ones we don't need. You'll probably need some adjustments to `getStyles()`, but here's a rough example of what I mean:

```typescript
const BaseButton = props => {
    const styles = StyleSheet.create(getStyles(props.globalButtonSettings))
    props.href ? <LinkButton {...props} className={styles.button} /> : <StandardButton {...props} className={styles.button} />
}

return (
    <BaseButton 
        {...props} 
        globalButtonSettings={
            props.buttonStyle === "primary" ? primaryButtonStyles : secondaryButtonStyles;
        }
)
```